lib/src/bindings/data_visitor.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:ffi';
+import 'signatures.dart';
+import "package:ffi/ffi.dart" show allocate, free;
+
+int _lastId = 0;
+final _callbacks = <int, bool Function(Pointer<Uint8> dataPtr, int length)>{};
```

Comment:
Lambdas can't be passed as a C callback, FFI requires static functions. I've tried that (lambda) approach but had to resolve to this trampoline one instead (basically the same thing we do in Go)

Reply:
There is no concurrent memory access in Dart. All code in a single isolate executes on a single event-loop and isolates don't share memory.

Reply:
Haven't tried that yet and I think it's just like threads in any other language (except you can't share variables inside dart). 

And yes tests like that make sense. We had an issue regarding isolates so I kinda related stuff like this to that issue, in my head. writing it down now #52

Reply:
how so? I don't see a problem

Reply:
> Callbacks from core happen on a different thread

They don't with objectbox-c - all calls are synchronous, executed on the same thread. This approach wouldn't work for other kinds of callbacks, where you register the callback and it may happen in the future (e.g. some kind of event listeners). Not sure how that could be handled with dart, if at all.

Reply:
Looks like it's an open issue with some workarounds available https://github.com/dart-lang/sdk/issues/37022